patmat/src/main/scala/patmat/Huffman.scala

```diff
@@ -124,7 +158,20 @@ trait Huffman extends HuffmanInterface:
    * This function decodes the bit sequence `bits` using the code tree `tree` and returns
    * the resulting list of characters.
    */
-  def decode(tree: CodeTree, bits: List[Bit]): List[Char] = ???
+  def decode(tree: CodeTree, bits: List[Bit]): List[Char] = {
+    def decodeInner(currentDecodeTree: CodeTree, bitsLeft: List[Bit]): List[Char] = {
+      currentDecodeTree match {
+        case Leaf(char, _) => char :: decodeInner(tree, bitsLeft)
+        case Fork(left, right, _, _) => bitsLeft match {
+          case 0 :: bs => decodeInner(left, bs)
+          case 1 :: bs => decodeInner(right, bs)
+          case Nil => Nil
```

Comment:
```suggestion
          case _ => Nil
```

Reply:
> 

I think it's better not to match everything to Nil, because we could have base-2 change to base-3 in the future, and then "2 :: bs" would be implicitly mapped to Nil, while it would be better to have an explicit error of match not handling this case